### PR TITLE
[ENH] enable the transfer of build infrastructure into `_build/myst` for buildable output

### PR DESCRIPTION
**Pull request by AakashGfude**
PR implements : 

- [x] copy conf.py to _build/myst
- [x] copy Makefile to _build/myst
- [x] update conf.py to include myst_parser

The update of `conf.py` is done using a simple string check for keywords. 
But I was hoping I would be able to load the file as a pickle or a module and add `myst_parser` to the `extensions` object directly. But the file is throwing error on pickling. And module editing is not working.

- [x]  mimic the source in the build directory

fixes https://github.com/mmcky/sphinxcontrib-rst2myst/issues/27

### CODE REVIEW

**Comment by mmcky** [comment]
@AakashGfude i don' think there is an easy way to convert `sphinx.app.config` back into the `conf.py` file so I think direct access for updating the `conf.py` is probably a good way to go for now.

**Comment by AakashGfude** [comment]
thanks, @mmcky. I will implement the update of `Makefile` and test cases for now then

File: sphinxcontrib/rst2myst/writers/accumulators.py [hunk 1]
```diff
@@ -100,7 +100,8 @@ def start_list_item(self):
         self.list_item = []
 
     def addto_list_item(self, content):
-        self.list_item.append(content)
+        if self.list_item:
```
Comment (mmcky) [inline sphinxcontrib/rst2myst/writers/accumulators.py:103]: @AakashGfude what is this condition for?

Comment (AakashGfude) [reply sphinxcontrib/rst2myst/writers/accumulators.py:103]: @mmcky on running `make myst` on https://github.com/QuantEcon/lecture-python-advanced . it was throwing error - `cannot append to NoneType` for `self.list_item`. So just checking if the list exists before appending. Is `addto_list_item` function always called after `start_list_item` has been called?

Comment (mmcky) [reply sphinxcontrib/rst2myst/writers/accumulators.py:103]: not sure -- but will look into it. I think state checking like this should probably be in the `translator` to avoid introduction of possible edge cases. This also seems to alter the `myst` output to fail tests. I will take a closer look. Thanks @AakashGfude

Comment (AakashGfude) [reply sphinxcontrib/rst2myst/writers/accumulators.py:103]: aah ok. true should transfer it to the translator where this function is called.

**Comment by mmcky** [comment]
> edit the copied Makefile to change the SOURCEDIR from source/rst to the relative path of .md files

alternatively should the `_build/myst` folder mirror the source folder?

**Comment by AakashGfude** [comment]
> > edit the copied Makefile to change the SOURCEDIR from source/rst to the relative path of .md files
> 
> alternatively should the `_build/myst` folder mirror the source folder?

you mean copying all the source rst files with their folder structure over to `_build/myst` ?

**Comment by mmcky** [comment]
@AakashGfude just copying the folder structure of the source files (not the `rst` files themselves) `:-)`

**Comment by mmcky** [comment]
@AakashGfude it might be more robust and a good choice to just substitute 

`sphinxcontrib.rst2myst` with `myst_parser` when adjusting the `conf.py` for `_build/myst`

**Comment by AakashGfude** [comment]
> @AakashGfude it might be more robust and a good choice to just substitute
> 
> `sphinxcontrib.rst2myst` with `myst_parser` when adjusting the `conf.py` for `_build/myst`

Yes. That's true

**Comment by mmcky** [comment]
thanks @AakashGfude I will merge this tomorrow and check it out.

**Comment by mmcky** [comment]
Is there anyway I can check out your branch without cloning your repo? I remember this is a way but just can't recall the instructions. I might just merge as is  -- and then test.
